### PR TITLE
Add authentication and comment feature

### PR DESCRIPTION
**Pull request by gaurnett**
Hi Aditya and Taylor, 

I added an authentication feature that allows users to sign in and post comments. Auth allows me to now let specific users delete their own comments.

http://gaurnett-step-2020.uc.r.appspot.com/

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/User.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if (isUserSignedIn) {
+            id = userService.getCurrentUser().getUserId();
+            name = userService.getCurrentUser().getNickname();
+            email = userService.getCurrentUser().getEmail();
+            link = userService.createLogoutURL("/");
+        } else {
+            link = userService.createLoginURL("/");
+        }
+    }
+}
```
Comment (taylorwan) [inline portfolio/src/main/java/com/google/sps/data/User.java:24]: nit: missing newline

File: portfolio/src/main/webapp/comments.html [hunk 2]
```diff
@@ -1,114 +1,97 @@
...
+    <script src="./js/jquery.js"></script>
+    <script src="./js/script.js"></script>
+</body>
 
-        <!-- Custom JS Files -->
-        <script src="./js/jquery.js"></script>
-        <script src="./js/script.js"></script>
-    </body>
-</html>
+</html>
```
Comment (taylorwan) [inline portfolio/src/main/webapp/comments.html:97]: nit: newline

File: portfolio/src/main/webapp/css/style.css [hunk 3]
```diff
@@ -152,14 +152,28 @@ section {
...
+    margin-right: 5px;
+}
+
+.show-element {
+    display: block;
+}
+
+.hide-element {
+    display: none;
+}
```
Comment (taylorwan) [inline portfolio/src/main/webapp/css/style.css:179]: nit: newline

**Review by taylorwan — APPROVED** [review]
Well done!

One thought - now that we have user authentication, what do you think about removing the "Full name" input on the comment form? Do we want to give users that option, or do we want to use their Gmail name by default?

**Comment by taylorwan** [comment]
Also, it looks like my first comment now says "undefined" - can we default to the full name field when there is no user data?

**Comment by gaurnett** [comment]
The first comment says undefined because I changed the name of the comment attribute from name to posterName. I deleted this comment from the datastore so this issue won't happen again. Also, this API does not give me access to fullname, only the username, so I have to continue asking for their name.

**Pull request merged**